### PR TITLE
CEN-34

### PR DESCRIPTION
**Pull request by Tajae1532**

### CODE REVIEW

**Review by eandrews737 — DISMISSED** [review]

**Review by lamad019 — APPROVED** [review]

**Review by lamad019 — DISMISSED** [review]
Conflicting files removed, should be good now

**Review by eandrews737 — APPROVED** [review]

File: djangoBookStoreApi/urls.py [hunk 1]
```diff
@@ -2,12 +2,17 @@
...
 # add route paths here
 urlpatterns = [
     path('test/', test_view),
     path('user/<int:id>/shopping_carts', shopping_carts_view),
     path('user/<int:user_id>/profile', UserProfileView.as_view(), name='user_profile'),
+    path('user/create', CreateUserView.as_view(), name='user-create'),
+    path('user/', UserDetailView.as_view(), name='user-detail'),
     path('user/<int:id>/wishlist', wish_list_books_view),
 ]
+#h
```
Comment (eandrews737) [inline djangoBookStoreApi/urls.py:18]:
```suggestion
```

File: djangoBookStoreApi/views/user_profile_view.py [hunk 2]
```diff
@@ -10,3 +12,61 @@ def get(self, request, user_id):
...
+                'message': 'User created successfully',
+                'user_id': user.id
+            }
+            return JsonResponse(response)
+        else:
+            response = {
+                'error': 'Username is required'
+            }
+            return JsonResponse(response, status=400)
+        #h
```
Comment (eandrews737) [inline djangoBookStoreApi/views/user_profile_view.py:56]:
```suggestion
```

File: djangoBookStoreApi/views/user_profile_view.py [hunk 3]
```diff
@@ -10,3 +12,61 @@ def get(self, request, user_id):
...
+        try:
+            data = json.loads(request.body)
+        except json.JSONDecodeError:
+            data = {}
+
+        # Extract username from the JSON data
+        username = data.get('username')
+
+        # Check if username is provided and not empty
+        if username:
```
Comment (eandrews737) [inline djangoBookStoreApi/views/user_profile_view.py:29]: we need to make checks for if they have username and password at the beginning of this function

**Review by eandrews737 — CHANGES_REQUESTED** [review]

**Comment by Tajae1532** [comment]
Will Close

**Pull request closed without merging**